src/concurrency/weak_memory.rs

```diff
@@ -108,6 +108,19 @@ pub struct StoreBufferAlloc {
     store_buffers: RefCell<RangeObjectMap<StoreBuffer>>,
 }
 
+impl StoreBufferAlloc {
+    pub fn iter(&self, mut visitor: impl FnMut(&Scalar<Provenance>)) {
```

Comment:
Why does this need to be `pub`? I'd expect a visitor impl somewhere inside `src/concurrency`.

Reply:
I don't know how to do this with a visitor impl, because this is "visited" in the implementation of the visitor. This is a part of `AllocExtra`, not a member of `Machine`.

Reply:
> This is a part of AllocExtra, not a member of Machine.

So what?
`machine::AllocExtra` should also implement the trait so that all the per-allocation state is being visited.

Reply:
As far as I understand, we call `this.machine.vist_all_machine_values` then from the `Machine` we recurse down to all the parts of `Machine`. I don't know how to recurse down into the allocations from the `Machine`.